### ISSUE TITLE
Test if we can start running a workflow while its input is partially staged 

### ISSUE THREAD

**Issue by haozturk** [CMSCompOps/WmAgentScripts#924]
**Impact of the new feature**
For now: B-parking staging and processing - later it can be used for other campaigns as well

**Is your feature request related to a problem? Please describe.**
B-parking staging is taking so long and we're considering the option of running the workflows w/ the already staged input. The problem is that we're not sure if the jobs whose input is not available will fail or not. 

**Describe the solution you'd like**
We'll submit a test workflow to understand how the system will perform. Here is the idea:

1. Pick a campaign which is not used right now.
2. Find a workflow from this campaign whose input is not on disk
3. Set the `partial_copy` parameter of the campaign to a small value such that the workflow can start running w/ partial input.
4. Submit the workflow as a backfill
5. Let MSTransferor create the input rules
6. Suspend some of the rules immediately and let other rules go
7. Let the workflow run with whatever input it has
8. Monitor if we're failing
9. If we're not failing - let the suspended rules go and monitor again.

This campaign looks like a good candidate as Jordan told me that it will not be used soon: https://dmytro.web.cern.ch/dmytro/cmsprodmon/requests.php?task_campaign=UltraLegacy2017 So we can play w/ its partial copy. The problem is that we should not pick a dataset whose tape copy at CERN. We don't want to mess up w/ the staging of B-parking as it's being staged from CERN tape.

**Describe alternatives you've considered**
None

**Additional context**
@z4027163  @jenimal  @drkovalskyi  @amaltaro  @todor-ivanov  fyi

**Comment by haozturk** [comment]
I'll use this workflow for testing: https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=ReReco-Run2017E-L1MinimumBias2-09Aug2019_UL2017_LowPU-00001 
Input is small (1.9TB) It has a copy at FNAL tape, it should be partially staged quickly. There are just 3 blocks, so we expect 3 rules. We can suspend 2 of them and see how it goes. I'll update the ticket after submitting the test.

**Comment by haozturk** [comment]
This is the test workflow: 
https://cmsweb.cern.ch/reqmgr2/fetch?rid=haozturk_Run2017E-v1_L1MinimumBias2_09Aug2019_UL2017_LowPU-Backfill_210903_182245_6064 
It has 3 Rucio rules: 2 of them is suspended and 1 of them is replicating. 
Partial copy is 10%: https://cmsweb.cern.ch/reqmgr2/data/campaignconfig/UltraLegacy2017-Backfill 
Monitor staging from here: https://cmsweb.cern.ch/ms-monitor/data/info?request=haozturk_Run2017E-v1_L1MinimumBias2_09Aug2019_UL2017_LowPU-Backfill_210903_182245_6064
or w/ `rucio rule-info 03b1562d1c764d068d71129363beb21f `

**Comment by haozturk** [comment]
it seems that SUSPENDED rules didn't stop the transfers... We'll delete the rules and also the files and try it again w/ another workaround.

**Comment by haozturk** [comment]
This is the new test workflow: https://cmsweb.cern.ch/reqmgr2/fetch?rid=haozturk_Run2017E-v1_L1MinimumBias2_09Aug2019_UL2017_LowPU-Backfill_210907_110316_6025
This time, we manually created 1 rule which will stage ~20% of the input. We bypassed MSTransferor such that it'll not be able to create new rules. Let's see how this test will go. 

Partial staging request:
`rucio rule-info 65718228ae8b451b83c8feb2c3314c70`

**Comment by haozturk** [comment]
Rule is successful, so only 23% of the input is on disk. Partial copy is 10%. Manually setting the workflow status to `staged`. Let's see how the rest will go. @z4027163  @jenimal  FYI

**Comment by z4027163** [comment]
It seems the WF works fine, it has 20% completion.

**Comment by haozturk** [comment]
I agree, no failure. It seems agents didn't spawn jobs for input which is not available yet, which is exactly what we needed. @todor-ivanov  @amaltaro according to this test, it seems that we can start processing the B-parking w/ partial input after setting `partial_copy` in campaign level. Do you also agree? Do you see any potential issues?

What I want to right now is to stage 1 more input block and monitor how it will be processed. This way the scenario will be more complete and we can be more confident.

**Comment by amaltaro** [comment]
Workqueue elements are looking sane for me:
https://cmsweb.cern.ch/couchdb/workqueue/_design/WorkQueue/_rewrite/elementsInfo?request=haozturk_Run2017E-v1_L1MinimumBias2_09Aug2019_UL2017_LowPU-Backfill_210907_110316_6025

I say we are good to go. Thanks for running this test, Hasan!

**Comment by haozturk** [comment]
Felipe started staging of another input block. Let's see how it'll go:
`rucio rule-info c6a9a32ca1ee4e7f9c5ca2c7b9ed1dfd`
